xmtp_mls/src/groups/mod.rs

```diff
@@ -706,6 +714,36 @@ impl<ScopedClient: ScopedGroupClient> MlsGroup<ScopedClient> {
         Ok(message_id)
     }
 
+    /// Helper function to extract queryable content fields from a message
+    fn extract_queryable_content_fields(message: &[u8]) -> QueryableContentFields {
+        let default = QueryableContentFields {
+            content_type: ContentType::Unknown,
+            version_major: 0,
+            version_minor: 0,
+            authority_id: "unknown".to_string(),
+        };
+
+        // Return early with default if decoding fails or type is missing
+        let content_type_id = match EncodedContent::decode(message)
+            .map_err(|e| tracing::debug!("Failed to decode message as EncodedContent: {}", e))
+            .ok()
+            .and_then(|content| content.r#type)
+        {
+            Some(type_id) => type_id,
+            None => {
+                tracing::debug!("Message content type is missing");
```

Comment:
This matching on Some/None is redundant, since its just the transformed error we're already throwing a log on